initrd/init

```diff
@@ -67,11 +51,12 @@ if [ ! -z "$CONFIG_USB_BOOT_DEV" ]; then
 	echo >> /etc/fstab "$CONFIG_USB_BOOT_DEV /media auto defaults,ro 0 0"
 fi
 
-if [ ! -x "$CONFIG_BOOTSCRIPT" ]; then
-	recovery 'Boot script missing?  Entering recovery shell'
-	# just in case...
-	tpm extend -ix 4 -ic recovery
-	exec /bin/ash
+# Setup recovery serial shell
+if [ ! -z "$CONFIG_BOOT_RECOVERY_SERIAL" ]; then
+	stty -F "$CONFIG_BOOT_RECOVERY_SERIAL" 115200
+	pause_recovery 'Console recovery shell' \
```

Comment:
Does this require user intervension on every boot with `CONFIG_BOOT_RECOVERY_SERIAL` set?

Reply:
Nope - you can boot normally via generic-init through the internal serial connected to the BMC. This is just here to enable recovery shell access on the external serial port on the back of the motherboard (in cases when the BMC is not active or down). This allows you to flash openbmc when physically local to the machine. You can also do this remotely if you configure network-init-recovery as the network bootscript (same drop to recovery shell on ssh into the machine)